source/vision/visionHandler.py

```diff
@@ -292,26 +292,24 @@ def handleMouseMove(self, obj, x: int, y: int) -> None:
 		self.extensionPoints.post_mouseMove.notify(obj=obj, x=x, y=y)
 
 	def handleConfigProfileSwitch(self) -> None:
-		configuredProviders = set(config.conf['vision']['providers'])
-		curProviders = set(self._providers)
+		configuredProviders: Set[providerInfo.ProviderInfo] = set(self.getConfiguredProviderInfos())

```

Comment:
If taking this approach, I'd like some documentation about how the equivalence tests work here, or at least why we don't have to worry about it, and how to ensure it's not broken in the future. My initial guess is that we don't have to worry too, the root source of the objects for both `configuredProviders` and `curProviders` is `_allProviders`. Though this isn't immediately obvious, and may be broken by an extension to the `ProviderInfo`?

Actually, I'm really not sure it's helpful to create these sets of `ProviderInfo` objects, rather than getting the `ProviderInfo` object before terminate / initialize.